### DISCUSSION TITLE
Publish a Milestone of Grails 7

### DISCUSSION THREAD

**Discussion by sdelamo** [apache/grails-core#13561] [Ideas]
I think we could release Grails 7.0.0-M1 as an artificial for internal consumption. Thanks to @matrei work, grails-core 7.0.x branch now builds with [Spring Boot 3.2 and Micronaut 4](https://github.com/grails/grails-core/pull/13545), and [Apache Groovy 4.0](https://github.com/grails/grails-core/pull/13532). 

Once, we have a 7.0.0-M1 artefact, we could create major branches in gsp, scafolding etc and try to get the main plugins with the latest. 

Once the main plugins are building correctly, I think we should move then into core. 

Thoughts?

**Comment by matrei** [comment]
Sounds good to me :+1:

**Comment by wuwu2000** [comment]
Hello, do you have any update plans when to release grails 7?
Since spring framework 5.3 is EOL since August and lately CVE-2024-38819 and CVE-2024-38820 (https://spring.io/blog/2024/10/17/spring-framework-cve-2024-38819-and-cve-2024-38820-published) have been published, they are only fixed in commercial version, the grails 7/spring 6 update seems more pressuring then ever

**Reply by codeconsole** [reply]
@wuwu2000 7.0.0-M1 should be coming end of this month with the release of Spring Boot 3.4.0 
The 7.0.0 stack will be based off of 3.4.x

**Reply by jeffscottbrown** [reply]
"7.0.0-M1 should be coming end of this month" - Good news.  Well done to the development team!

**Reply by wuwu2000** [reply]
thank you for your work, I am looking forward to it!

**Comment by jeffscottbrown** [comment]
"I think we could release Grails 7.0.0-M1 as an artificial for internal consumption." - Are there good reasons not to make M1 public?

**Reply by codeconsole** [reply]
not sure what that is, but IMO, the plan would be to make it public under the condition it is a milestone and subject to major changes that bypass the deprecation process.

**Reply by matrei** [reply]
I think this was written quite early in the process, when it could have been good to have something solid to start migrating other core plugins and projects towards. As this did not happen we have been building against snapshots instead.

But now we are in a much better position to release a "public" M1. I think it would be good if we could focus on clearing any `@PendingFeature` or `@Ignore` annotated tests in `grails-functional-tests` (that were added for Grails 7) before M1 is released, because there are some showstoppers there.

**Reply by jeffscottbrown** [reply]
"But now we are in a much better position to release a "public" M1."  +1
